### PR TITLE
[WIP] Add useEvent

### PR DESCRIPTION
**Pull request by rickhanlonii**
## Overview

This PR adds a new hook `useEvent` per the RFC [here](https://github.com/reactjs/rfcs/pull/220).

## TODO

- [ ] More tests
- [ ] Mutate in the correct time of the commit phase (currently just an insertion effect)
- [ ] Decide on the error message text for calling it in render
- [ ] Update linter plugin treat the useEvent return values in scope as “stable”
- [ ] Figure out the correct type
- [ ] Add DevTools support

### CODE REVIEW

File: packages/react-reconciler/src/__tests__/useEvent-test.js [hunk 1]
```diff
@@ -0,0 +1,406 @@
...
+ *
+ * @emails react-core
+ * @jest-environment node
+ */
+
+/* eslint-disable no-func-assign */
+
+'use strict';
+
+describe('useRef', () => {
```
Comment (scottrippey) [inline packages/react-reconciler/src/__tests__/useEvent-test.js:15]:
```suggestion
describe('useEvent', () => {
```

File: packages/react-reconciler/src/ReactFiberWorkLoop.old.js [hunk 2]
```diff
@@ -1354,6 +1354,10 @@ export function getExecutionContext(): ExecutionContext {
   return executionContext;
 }
 
+export function isCurrentlyRendering(): boolean {
```
Comment (incepter) [inline packages/react-reconciler/src/ReactFiberWorkLoop.old.js:1357]: unused fn ?

File: packages/react-reconciler/src/__tests__/useEvent-test.js (hunk 1 above)
Comment (rickhanlonii) [reply packages/react-reconciler/src/__tests__/useEvent-test.js]: Thanks, fixed!

**Comment by rickhanlonii** [comment]
Finished in https://github.com/facebook/react/pull/25229

**Pull request closed without merging**